### DISCUSSION TITLE
2021-08-31 weekly planning

### DISCUSSION THREAD

**Discussion by kgodey** [mathesar-foundation/mathesar#614] [Maintainer discussions]
### Discussion topics

- **[Updates](https://github.com/centerofci/mathesar/discussions/614#discussioncomment-1259664)**: General updates everyone should be aware of.
- **[Questions](https://github.com/centerofci/mathesar/discussions/614#discussioncomment-1259665)**: Questions for the team to answer.
- **[Active Work](https://github.com/centerofci/mathesar/discussions/614#discussioncomment-1259675)**: Work planned for the next week.
- **[Retrospective](https://github.com/centerofci/mathesar/discussions/614#discussioncomment-1259684)**: Summary of last week's work.

### About
Each Tuesday, we start a discussion about Mathesar's progress. Everyone is welcome to add a new topic to discuss or to comment on existing topics.

This post covers two weeks since we didn't do a discussion last week.

**Comment by kgodey** [comment]
## Updates

### Progress
We completed 3 milestones in the last 2 weeks!
- [03. Working with Schemas](https://github.com/centerofci/mathesar/milestone/6) - 14 issues
- [04. Tables from File Import](https://github.com/centerofci/mathesar/milestone/1) - 33 issues
- [2021-08 improvements](https://github.com/centerofci/mathesar/milestone/29) - 23 issues

We've also now completed over 200 issues in total.

### Roadmap Updates
Based on progress so far, we need to cut some features from our plan to ensure that we release a high-quality alpha version of Mathesar by the end of 2021. Our roadmap was fairly ambitious, so this is not a surprise. The Mathesar core team is meeting tomorrow to discuss the plan, but I made some initial changes to our roadmap to get us started. The plan is to focus on getting the experience of working with tabular data right for the alpha release before implementing complex new features like visualizations and real-time collaboration.

Please review the [updated roadmap](https://github.com/centerofci/mathesar/milestones?direction=asc&sort=due_date&state=open) to see the latest state. We also added two new milestones to focus on stability - milestones [06](https://github.com/centerofci/mathesar/milestone/34) and [13](https://github.com/centerofci/mathesar/milestone/35). This will give us some decided time to address technical debt and fix issues with what we've built up to that point.

### Core Team
- @eito-fis is done with his internship and is back to college. Thanks for all of your hard work, it was a pleasure to work with you!
- @mathemancer is back from parental leave.

### Contributors
We have a couple of new contributors who had their first pull requests merged into Mathesar. Welcome to @asharonbaltazar, @linzhou-zhong, and @arvind-1. :)

Also, @mr-gabe49 had a couple more pull requests merged in, nice work!

**Reply by eito-fis** [reply]
Thanks, I learned a ton this summer! And I'll still be around to help with reviews and PRs :slightly_smiling_face:

**Comment by kgodey** [comment]
## Questions

@mathemancer, @powellc, @ghislaineguerin, @pavish: Your responses are required. Others are also welcome to respond.

### Retrospective
- What went well over the past two weeks?
- What could have gone better over the past two weeks?
- Are there any improvements to our process you'd like to suggest?

### Active Work
- Does your workload for this week look okay?
- We have one milestone due in the next week. Are there any blockers to meeting the deadline?

### Discussion
Please respond to the [Updates](https://github.com/centerofci/mathesar/discussions/614#discussioncomment-1259664) post directly if you'd like to discuss anything there. If you have any other topics to discuss, please post them as new replies to the [parent post](https://github.com/centerofci/mathesar/discussions/614#discussion-3548369).

**Reply by kgodey** [reply]
> What went well over the past two weeks?

With my work: I closed all the API bugs that Pavish opened, some of which involved major refactoring. I am more comfortable with the `db` code now.

With the project: We closed three milestones and we have more contributors. :). It's especially nice to get the tables from file import done, it was one of the first things we started work on.

> What could have gone better over the past two weeks?

I can't think of anything specific. It took me longer to refactor the `db` code than I anticipated, but I'm not sure that could have been helped.

> Are there any improvements to our process you'd like to suggest?

I think pausing feature work to do some refactoring is a good idea at this point, I'm glad we're doing it. But that's something we're already planning to do.

> Does your workload for this week look okay?

I probably won't get everything done, I have 7 issues assigned to me, but they're mostly all refactoring related, so I think it makes sense to think about them together.

> We have one milestone due in the next week. Are there any blockers to meeting the deadline?

Not from me.

**Reply by mathemancer** [reply]
> What went well over the past two weeks?

N/A

> What could have gone better over the past two weeks?

N/A

> Are there any improvements to our process you'd like to suggest?

Nothing I can think of at the moment

> Does your workload for this week look okay?

I think so, though the mathesar `MONEY` type is a bit more complicated than I originally thought.

> We have one milestone due in the next week. Are there any blockers to meeting the deadline?

Nothing I can think of.

**Reply by powellc** [reply]
> What went well over the past two weeks?

Finally getting comfortable with the full dev process, the code base, and especially the DB reflection. First PR merged and I'm hoping my development speed ramps up nicely from here.

> What could have gone better over the past two weeks?

I need to be opening PRs and reaching out for comment on my solutions sooner. I think I probably could have been working faster the last week if I'd just opened the PR instead of sitting on the solution for a few days.

I also am still a little time limited due to personal obligations, so figuring out the best days and times to get focused work done can be improved, and then communicating that to the team.

> Are there any improvements to our process you'd like to suggest?

Not at the moment.

> Does your workload for this week look okay?

Yes. With a format taking shape for documenting filtering, and a good test frame for filtering types in the API, I should move faster through those tickets this week and be able to spend some time on Friday going deep on #248 

> We have one milestone due in the next week. Are there any blockers to meeting the deadline?

Not from me.

**Reply by ghislaineguerin** [reply]
> What went well over the past two weeks?

Global documents/prototypes seem to be working well to resolve the issues blocking the review process. I'm also getting a lot from doing quick design reviews w/ Kriti.

> What could have gone better over the past two weeks?

Nothing specific. I think it went well in general.

> Are there any improvements to our process you'd like to suggest?

Continue to do global documents at the end of a milestone. So, for example, once views are complete, have a prototype covering all prototypes for that milestone.

> Does your workload for this week look okay?

It does.

> We have one milestone due in the next week. Are there any blockers to meeting the deadline?

Not that I'm aware of.

**Reply by pavish** [reply]
> What went well over the past two weeks?

We're progressing on completing milestones. There were more external contributions for the frontend. Frontend work is moving in a steady pace.

> What could have gone better over the past two weeks?

Didn't get to work much last week, due to health concerns. I think overall, things went well during the past two weeks.

> Are there any improvements to our process you'd like to suggest?

Nothing at the moment.

> Does your workload for this week look okay?

Yes, it looks good.

> We have one milestone due in the next week. Are there any blockers to meeting the deadline?

None that I can think of.

**Comment by kgodey** [comment]
## Active Work

This post lists work planned for this week (see also: [Active](https://github.com/centerofci/mathesar/projects/1)). It also includes a current summary of milestones involved in this week's work.

### [2021-09-07] [05. Editable Tables](https://github.com/centerofci/mathesar/milestone/4)

- **[8 issues](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+is%3Aclosed+milestone%3A%2205.%20Editable%20Tables%22) already closed.** 
- **Issues currently in progress:**
  - [@pavish] #195
- **Issues not yet started:**
  - [@pavish] #197

### [2021-09-20] [06. 2021-09 Stability](https://github.com/centerofci/mathesar/milestone/34)

- **Issues currently in progress:**
  - [@andyharley-cci] #225
- **Issues not yet started:**
  - [@kgodey] #280
  - [@kgodey] #582
  - [@kgodey] #583
  - [@kgodey] #592
  - [@kgodey] #510
- **Issues remaining in backlog:**
  - 3 [backend](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20backend%22+milestone%3A%2206.%202021-09%20Stability%22) and [database](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20database%22+milestone%3A%2206.%202021-09%20Stability%22) issues
  - 1 [frontend](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20frontend%22+milestone%3A%2206.%202021-09%20Stability%22) issues
  - 1 [design](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20design%22+milestone%3A%2206.%202021-09%20Stability%22) issues

### [2021-09-27] [07. Working with Tables](https://github.com/centerofci/mathesar/milestone/8)

- **[19 issues](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+is%3Aclosed+milestone%3A%2207.%20Working%20with%20Tables%22) already closed.** 
- **Issues currently in progress:**
  - [@raduab] #328
  - [@seancolsen] #444
  - [@dmos62] #498
  - [@pegnott] #495
- **Issues not yet started:**
  - [@Lucasmiguelmac] #500
- **Issues remaining in backlog:**
  - 10 [frontend](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20frontend%22+milestone%3A%2207.%20Working%20with%20Tables%22) issues

### [2021-09-30] [2021-09 improvements](https://github.com/centerofci/mathesar/milestone/30)

- **[2 issues](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+is%3Aclosed+milestone%3A%222021-09%20improvements%22) already closed.** 
- **Issues currently in progress:**
  - [@asharonbaltazar] #594
  - [@arvind-1] #555
  - [@linzhou-zhong] #517
  - [@andyharley-cci] #421
  - [@mr-gabe49] #609
- **Issues remaining in backlog:**
  - 10 [frontend](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20frontend%22+milestone%3A%222021-09%20improvements%22) issues
  - 3 [infrastructure](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20infrastructure%22+milestone%3A%222021-09%20improvements%22) issues

### [2021-10-04] [08. Initial Data Types](https://github.com/centerofci/mathesar/milestone/11)

- **[26 issues](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+is%3Aclosed+milestone%3A%2208.%20Initial%20Data%20Types%22) already closed.** 
- **Issues currently in progress:**
  - [@powellc] #387
  - [@mathemancer] #417
  - [@eito-fis] #426
  - [@powellc] #388
- **Issues not yet started:**
  - [@kgodey] #552
  - [@kgodey] #392
  - [@rootxnet] #404
  - [@rootxnet] #405
  - [@rootxnet] #407
  - [@powellc] #384
  - [@powellc] #385
  - [@mathemancer] #412
  - [@mathemancer] #424
  - [@mathemancer] #430
- **Issues remaining in backlog:**
  - 25 [backend](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20backend%22+milestone%3A%2208.%20Initial%20Data%20Types%22) and [database](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20database%22+milestone%3A%2208.%20Initial%20Data%20Types%22) issues
  - 8 [frontend](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20frontend%22+milestone%3A%2208.%20Initial%20Data%20Types%22) issues

### [2021-10-18] [09. Working with Views](https://github.com/centerofci/mathesar/milestone/13)

- **[4 issues](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+is%3Aclosed+milestone%3A%2209.%20Working%20with%20Views%22) already closed.** 
- **Issues currently in progress:**
  - [@ghislaineguerin] #466
  - [@ghislaineguerin] #454
  - [@ghislaineguerin] #243
- **Issues not yet started:**
  - [@ghislaineguerin] #455
  - [@ghislaineguerin] #456
  - [@ghislaineguerin] #443
- **Issues remaining in backlog:**
  - 7 [backend](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20backend%22+milestone%3A%2209.%20Working%20with%20Views%22) and [database](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20database%22+milestone%3A%2209.%20Working%20with%20Views%22) issues
  - 1 [frontend](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20frontend%22+milestone%3A%2209.%20Working%20with%20Views%22) issues
  - 1 other issues

### [2021-11-01] [11. Multiple Database Support](https://github.com/centerofci/mathesar/milestone/5)

- **[7 issues](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+is%3Aclosed+milestone%3A%2211.%20Multiple%20Database%20Support%22) already closed.** 
- **Issues currently in progress:**
  - [@aaditkamat] #395
- **Issues remaining in backlog:**
  - 4 [backend](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20backend%22+milestone%3A%2211.%20Multiple%20Database%20Support%22) and [database](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20database%22+milestone%3A%2211.%20Multiple%20Database%20Support%22) issues
  - 3 [frontend](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20frontend%22+milestone%3A%2211.%20Multiple%20Database%20Support%22) issues
  - 2 [design](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20design%22+milestone%3A%2211.%20Multiple%20Database%20Support%22) issues

**Comment by kgodey** [comment]
## Retrospective

This post lists issue updates from 2021-08-17 until now by milestone.

### [03. Working with Schemas](https://github.com/centerofci/mathesar/milestone/6)

- **Issues closed:**
  - [@pavish] #518

### [04. Tables from File Import](https://github.com/centerofci/mathesar/milestone/1)

- **Issues closed:**
  - [@pavish] #568
  - [@kgodey] #562
  - [@pavish] #295

### [2021-08 improvements](https://github.com/centerofci/mathesar/milestone/29)

- **Issues closed:**
  - [@kgodey] #595
  - [@kgodey] #579
  - #577
  - [@kgodey] #576
  - [@linzhou-zhong] #572
  - [@kgodey] #566
  - [@kgodey] #564
  - [@kgodey] #563
  - [@eito-fis] #559
  - [@pavish] #528
  - [@kgodey] #358
  - [@Dhruvi16] #334
  - [@Dhruvi16] #333
  - [@Dhruvi16] #332
  - [@Dhruvi16] #331
  - [@Dhruvi16] #330
  - [@Dhruvi16] #322
  - [@arvind-1] #313
- **New issues created:**
  - #595, created by @pavish
  - #579, created by @pavish
  - #577, created by @aaditkamat
  - #576, created by @pavish
  - #572, created by @eito-fis

### [05. Editable Tables](https://github.com/centerofci/mathesar/milestone/4)

- **Issues started, not yet finished:**
  - [@pavish] #195

### [06. 2021-09 Stability](https://github.com/centerofci/mathesar/milestone/34)

- **Issues started, not yet finished:**
  - [@andyharley-cci] #225
- **New issues created:**
  - #602, created by @kgodey
  - #597, created by @kgodey
  - #592, created by @kgodey
  - #583, created by @kgodey
  - #582, created by @kgodey

### [07. Working with Tables](https://github.com/centerofci/mathesar/milestone/8)

- **Issues closed:**
  - [@eito-fis] #515
  - [@eito-fis] #501
  - [@eito-fis] #494
  - [@eito-fis] #452
- **Issues started, not yet finished:**
  - [@raduab] #328
  - [@seancolsen] #444
  - [@dmos62] #498
  - [@pegnott] #495

### [2021-09 improvements](https://github.com/centerofci/mathesar/milestone/30)

- **Issues closed:**
  - [@mr-gabe49] #483
  - [@mr-gabe49] #474
- **Issues started, not yet finished:**
  - [@asharonbaltazar] #594
  - [@arvind-1] #555
  - [@linzhou-zhong] #517
  - [@andyharley-cci] #421
  - [@mr-gabe49] #609
- **New issues created:**
  - #609, created by @mr-gabe49
  - #608, created by @pavish
  - #594, created by @asharonbaltazar
  - #570, created by @asharonbaltazar

### [08. Initial Data Types](https://github.com/centerofci/mathesar/milestone/11)

- **Issues closed:**
  - [@ghislaineguerin] #537
  - [@eito-fis] #425
  - [@ghislaineguerin] #241
  - [@ghislaineguerin] #240
  - [@ghislaineguerin] #239
  - [@ghislaineguerin] #238
  - [@ghislaineguerin] #237
  - [@ghislaineguerin] #236
  - [@ghislaineguerin] #235
- **Issues started, not yet finished:**
  - [@powellc] #387
  - [@mathemancer] #417
  - [@eito-fis] #426
  - [@powellc] #388

### [09. Working with Views](https://github.com/centerofci/mathesar/milestone/13)

- **Issues started, not yet finished:**
  - [@ghislaineguerin] #466
  - [@ghislaineguerin] #454
  - [@ghislaineguerin] #243

### [10. Data Modeling](https://github.com/centerofci/mathesar/milestone/24)

- **Issues closed:**
  - [@kgodey] #538

### [2021-10 improvements](https://github.com/centerofci/mathesar/milestone/31)

- **New issues created:**
  - #603, created by @kgodey

### [11. Multiple Database Support](https://github.com/centerofci/mathesar/milestone/5)

- **Issues started, not yet finished:**
  - [@aaditkamat] #395

### [API v1](https://github.com/centerofci/mathesar/milestone/21)

- **New issues created:**
  - #593, created by @kgodey
